tanzu-build-service/troubleshooting.hbs.md

```diff
@@ -68,3 +68,30 @@ Tanzu Application Platform uses the default storage class which uses EBS volumes
 
 Follow the AWS documentation to install the [Amazon EBS CSI driver](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html)
 before installing Tanzu Application Platform, or before upgrading to Kubernetes v1.23.
+
+## Smart-warmer-image-fetcher reports ErrImagePull due to dockerd's layer depth limitation
+
+### Symptom
+
+When using dockerd as the cluster's container runtime, you might see the smart-warmer-image-fetcher pods 
+report a status of `ErrImagePull`. The cause of this error could be due to dockerd's layer depth limitation, 
+in which the maximum supported image layer depth is 125.
+
+### Explanation
+
+To verify that the `ErrImagePull` status is due to dockerd's maximum supported image layer depth, check for event 
+messages containing the words "max depth exceeded":
+
+   ```console
+   $ kubectl get events -A | grep "max depth exceeded"
+     build-service        73s         Warning     Failed         pod/smart-warmer-image-fetcher-wxtr8     Failed to pull image 
+     "harbor.somewhere.com/aws-repo/build-service:clusterbuilder-full@sha256:065bb361fd914a3970ad3dd93c603241e69cca214707feaa6
+     d8617019e20b65e":  rpc error: code = Unknown desc = failed to register layer: max depth exceeded
+   ```
+
+If you see the words, "max depth exceeded", you have likely encountered dockerd's maximum image layer depth limitation.
+
+### Solution
+
+To work around this issue, configure your cluster to use containerd as its default container runtime. Please refer to 
+your Kubernetes cluster provider's documentation for specific instructions.
```

Comment:
should we add link to the default documentation  for most used providers?

Reply:
Good idea! Added.